src/patchelf.cc

```diff
@@ -512,8 +516,8 @@ void ElfFile<ElfFileParamNames>::shiftFile(unsigned int extraPages, size_t start
     wri(phdr.p_offset, phdrs.at(splitIndex).p_offset - splitShift - shift);
     wri(phdr.p_paddr, phdrs.at(splitIndex).p_paddr - splitShift - shift);
     wri(phdr.p_vaddr, phdrs.at(splitIndex).p_vaddr - splitShift - shift);
-    wri(phdr.p_filesz, wri(phdr.p_memsz, splitShift + shift));
-    wri(phdr.p_flags, PF_R | PF_W);
+    wri(phdr.p_filesz, wri(phdr.p_memsz, splitShift + extraBytes));
+    wri(phdr.p_flags, splitFlags);
```

Comment:
This mades sense to me initially too but can be incorrect, since we can move a `.dynamic` section into here which needs write permissions, hence the hard RW. I suppose ideally we'd split this more fine grained, but that's a more complex fix.

That might be the cause of some CI failures.

Reply:
OK, that makes sense. I am admittedly not an expert in the intricacies of the ELF format, so I wasn't aware of this. When I'm able to look at this again (maybe not until next week), I will back out the change to the flags.

Reply:
I reverted the change to how the segment flags are handled. I'm not sure how to re-run the CI build though.